File: 0624-maximum-distance-in-arrays/0624-maximum-distance-in-arrays.java
class Solution {
    public int maxDistance(List<List<Integer>> arrays) {
        int smallest = 10001, biggest = -10001, maxDist = 0;
        for(var arr : arrays){
            maxDist = Math.max(maxDist, biggest - arr.get(0));
            maxDist = Math.max(maxDist, arr.get(arr.size() - 1) - smallest);
            smallest = Math.min(smallest, arr.get(0));
            biggest = Math.max(biggest, arr.get(arr.size() - 1));
        }

        return maxDist;
    }
}

File: 0624-maximum-distance-in-arrays/0624-maximum-distance-in-arrays.cpp
class Solution {
public:
    int maxDistance(vector<vector<int>>& arrays) {
        auto smallest = 10001, biggest = -10001, maxDist = 0;
        for(auto& arr : arrays){
            maxDist = std::max(maxDist, biggest - arr[0]);
            maxDist = std::max(maxDist, arr[arr.size()-1] - smallest);
            smallest = std::min(smallest, arr[0]);
            biggest = std::max(biggest, arr[arr.size()-1]);
        }

        return maxDist;
    }
};